contrib/sentence_transformers/test_sentence_transformers.py

```diff
@@ -22,6 +24,9 @@ def test_from_pretrained(multi_qa_model: SentenceTransformer) -> None:
     print("Similarity:", util.dot_score(query_embedding, passage_embedding))
 
 
-@pytest.mark.xfail(reason="Production endpoint is hardcoded in sentence_transformers when pushing to Hub.")
-def test_push_to_hub(multi_qa_model: SentenceTransformer, repo_name: str, cleanup_repo: None) -> None:
-    multi_qa_model.save_to_hub(repo_name)
+def test_push_to_hub(multi_qa_model: SentenceTransformer, repo_name: str, user: str, cleanup_repo: None) -> None:
+    multi_qa_model.save_to_hub(repo_name, organization=user)
+
+    # Check model has been pushed properly
+    model_id = f"{user}/{repo_name}"
+    assert model_info(model_id).library_name == "sentence-transformers"
```

Comment:
It looks like for some reason the `library_name` attribute [is not set when requesting the model info](https://github.com/huggingface/huggingface_hub/actions/runs/7246297614/job/19738349482?pr=1918) :confused:

Reply:
Maybe related https://github.com/huggingface/moon-landing/issues/8277

Reply:
Certainly odd. When I install with `git+https://github.com/UKPLab/sentence-transformers.git#egg=sentence-transformers`, it fails the first 2 times that I run `pytest`, and passes the 2 subsequent times :/

Reply:
When it fails, the only `siblings` in `ModelInfo` is `[RepoSibling(rfilename='.gitattributes', size=None, blob_id=None, lfs=None)]`, so the repo only has the gitattributes. If I `breakpoint` and inspect the times that it does pass, then all files are seen by the `model_info`.

Reply:
and between each run, is `model_id = f"{user}/{repo_name}"` changing? The value should be unique for each run (generated [here](https://github.com/huggingface/huggingface_hub/blob/a0716554e6d65f25565b69d24a0b1c2d62debc56/contrib/conftest.py#L33)). I'm asking because having dependency between tests is odd

Reply:
Yes, it does differ between runs.

Reply:
😵‍💫

Reply:
Maybe set a `time.sleep(1)` just before the `modle_info` call? It's not satisfying but it might have to do with some caching issue.

Reply:
I was thinking about that as well. Let me try.

Reply:
Haha, now the spaCy tests are failing & sentence_transformers is passing. Perhaps it's indeed a flaky test setup where `model_info` gets slightly outdated data if it is called "too soon".

Reply:
cc @Kakulukian @coyotte508 do you know if something has changed recently on the `/api/models/<repo_id>` endpoint that would make the model cache longer to update? In the test and thread above, we are doing:
1. create new repo
2. push model with modelcard
3. get model info (GET `/api/models/repo_id`)
4. check `model_info.library_name`.

It looks like adding a 1 second delay between steps 2. and 3. makes the test more robust. But I don't remember this was the case before. It is not a problem to add this delay in our tests but prefer to let you know in case it's a bigger problem server-side.